File: blog_vue/src/model/comment/CommentParent.ts
import CommentChild from "./CommentChild";
import CommentInformation from "./CommentInformation";

export default class CommentParent extends CommentInformation {
  commentChildList: Array<CommentChild>;
  childCount: number;

  constructor() {
    super();
    this.commentChildList = [new CommentChild()];
    this.childCount = -1;
    console.log(this.id);
  }
}

File: blog_springBoot/src/main/java/com/synergism/blog/core/comment/entity/CommentParent.java
package com.synergism.blog.core.comment.entity;

import lombok.Getter;
import lombok.Setter;

import java.util.List;

/**
 * 父评论信息
 */
@Setter
@Getter
public class CommentParent extends CommentInformation{
    //子评论列表
    private List<CommentChild> commentChildList;
    //子评论数
    private int childCount;

    public CommentParent(CommentInformation commentInformation, List<CommentChild> commentChildList,int childCount) {
        super(commentInformation);
        this.commentChildList = commentChildList;
        this.childCount = childCount;
    }
}
